app/3.0/docs/devguide/dom-template.md

```diff
@@ -4,169 +4,271 @@ title: DOM templating
-</dom-module>
+#### Provide template extension points {#insertcontent}
+
+Polymer makes it easy to provide template extension points in a base class, which a child
+class can then optionally override. You can provide template extension points by composing your
+base class template literal using expressions, like `${this.partialTemplate}`.
+
+The interpolated expressions act as partial templates ("partials") that the child class can override.
+
+```js
+...
+//create a base class template with extension points:
+static get template() {
+  return html`
+    <div>${this.headerTemplate}</div>
+    <p>Hello this is some content</p>
+    <div>${this.footerTemplate}</div>
+  `;
+}
+//supply the default content for the expressions:
+static get headerTemplate() { return html`...` }
+static get footerTemplate() { return html`...` }
+...
+```
+
+The expressions must return either a template element (`HTMLTemplateElement`) as shown above, or a specially-tagged
+string literal, using `htmlLiteral`:
+
+
+```js
+  static get stylePartial() {
+    return htmlLiteral`color: red`;
+  }
+```
+
+Your child class can override the extension points without changing
+the main template. Here's an example in which the child class overrides the base class content:
+
+Base class definition: { .caption }
+
+```js
+export class BaseClass extends Polymer.Element {
+  static get template() {
+    return html`
+      <div>${this.headerTemplate}</div>
+      <p>Hello this is some content</p>
+      <div>${this.footerTemplate}</div>
+    `;
+  }
+  static get headerTemplate() { return html`<h1>BaseClass: Header</h1>` }
+  static get footerTemplate() { return html`<h1>BaseClass: Footer</h1>` }
+}
 ```
 
+Child class definition: { .caption }
+```js
+export class ChildClass extends BaseClass {
+    // template definition inherited from BaseClass
+
+    // partial templates overridden by ChildClass
+    static get headerTemplate() { return html`<h2>ChildClass: Header</h2>` }
+    static get footerTemplate() { return html`<h2>ChildClass: Footer</h2>` }
+  }
+```
+
+[See a working example in Plunker](https://plnkr.co/edit/1uEd5Vv2Ob4bGHppzCs9?p=preview).
+
 ### Elements with no shadow DOM
 
-To create an element with no shadow DOM, don't specify a DOM template (either using a `<dom-module>`
-or by overriding the `template` getter), then no shadow root is created for the element.
+To create an element with no shadow DOM, don't specify a `template` getter. Then no shadow root is created for the element.
 
-If the element is extending another element that has a DOM template and you don't want a DOM template,
-define a `template` getter that returns a falsy value.
+If the element extends another element that has a DOM template, it will inherit that DOM template instead. To
+prevent the element from inheriting the superclass template, define a `template` getter that returns a falsy value:
+
+```js
+static get template() {
+  return null;
+}
+```
 
 ### URLs in templates {#urls-in-templates}
```

Comment:
### URLs in templates
 
I think some text needs updating in this section - it references HTML Imports

Reply:
Yeah, this is still sort of up in the air. I'll add a note here that the section needs to be updated when https://github.com/Polymer/polymer/issues/5163 gets resolved.